Bootstrap/app/src/main/java/ar/com/wolox/android/example/ui/logIn/LogInPresenter.java

```diff
@@ -1,24 +1,70 @@
 package ar.com.wolox.android.example.ui.logIn;
 
+import android.app.ProgressDialog;
 import android.content.Context;
 import android.content.SharedPreferences;
+import android.support.annotation.Nullable;
+import android.util.Log;
+import android.widget.EditText;
+
+import java.util.List;
 
 import javax.inject.Inject;
 
+import ar.com.wolox.android.R;
 import ar.com.wolox.android.example.TrainingApplication;
+import ar.com.wolox.android.example.model.User;
+import ar.com.wolox.android.example.network.PostService;
+import ar.com.wolox.android.example.network.UserService;
 import ar.com.wolox.wolmo.core.presenter.BasePresenter;
+import ar.com.wolox.wolmo.networking.retrofit.RetrofitServices;
+import ar.com.wolox.wolmo.networking.retrofit.callback.NetworkCallback;
+import butterknife.BindView;
+import okhttp3.ResponseBody;
+
+public class LogInPresenter extends BasePresenter<ILogInView> {
+    @Inject
+    RetrofitServices mRetrofitServices;
 
-class LogInPresenter extends BasePresenter<ILogInView> {
     private static final String SHARED_PREFERENCES="prefs";
     private static final String SHARED_PREFERENCES_KEY_EMAIL="mail";
+    private static final String PROGRESS_DIALOG_MESSAGE="Logging in...";
     @Inject
     public LogInPresenter(){
 
     }
-    public void doLogIn(String userName, String password, Context context) {
-        SharedPreferences sharedPref = context.getSharedPreferences(SHARED_PREFERENCES,Context.MODE_PRIVATE);
-        SharedPreferences.Editor editor = sharedPref.edit();
-        editor.putString(SHARED_PREFERENCES_KEY_EMAIL, userName);
-        editor.apply();
+    public void doLogIn(String email, String password, Context context) {
+        ProgressDialog pd = new ProgressDialog(context);
```

Comment:
The progressDialog should be in the Fragment. Remember that the Fragment decides what to show , and when.